src/librustc_session/options.rs

```diff
@@ -935,6 +935,8 @@ options! {DebuggingOptions, DebuggingSetter, basic_debugging_options,
         "run LLVM in non-parallel mode (while keeping codegen-units and ThinLTO)"),
     no_profiler_runtime: bool = (false, parse_no_flag, [TRACKED],
         "prevent automatic injection of the profiler_builtins crate"),
+    disable_unique_symbols: bool = (false, parse_bool, [UNTRACKED],
```

Comment:
Could this be changed to `enable_unique_symbols` or `use_unique_symbols`? I find it easier to deal with command-line arguments that don't involve double-negatives (e.g. "don't disable unique symbols").

Reply:
I understand the concern, but as far I can tell there's no double negative when passing `-Z disable-unique-symbols` to rustc in the command line. Is there another way to pass this?

Reply:
My concern is mainly when checking the flag - I think `!disable_unique_symbols` is slightly harder to understand than `enable_unique_symbols`. However, I don't have a strong opinion about it.

Reply:
The option may make sense if you have all the context of this PR, but is really ambiguous otherwise.
"Symbol" in particular is a very overloaded term.

I suggest changing the user-visible option to `-Z trim-diagnostic-paths`.
This way we make more obvious what this option is about and don't promise the specific algorithm used for the trimming.